app-editors/vis/vis-0.4.ebuild

```diff
@@ -3,21 +3,36 @@
 
 EAPI=5
 
+# Package test version and name
+MY_PTV=0.2
+MY_PTN=${PN}-test
+MY_PT=${MY_PTN}-${MY_PTV}
+
 DESCRIPTION="modern, legacy free, simple yet efficient vim-like editor"
 HOMEPAGE="https://github.com/martanne/vis"
-SRC_URI="https://github.com/martanne/vis/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI="https://github.com/martanne/vis/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	test? ( https://github.com/martanne/vis-test/archive/v${MY_PTV}.tar.gz -> ${MY_PT}.tar.gz )"
 LICENSE="ISC"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="+ncurses selinux tre"
+IUSE="+ncurses selinux test tre"
 
 #Note: vis is reported to also work with NetBSD curses
 #TODO: >=dev-lang/lua-5.2 (needed for syntax highlighting and settings)
-DEPEND=" dev-libs/libtermkey
+DEPEND=" app-eselect/eselect-vi
+	dev-libs/libtermkey
 	ncurses? ( sys-libs/ncurses:0= )
 	tre? ( dev-libs/tre:= )"
 RDEPEND="${DEPEND}"
 
+src_prepare() {
+	use test && (
```

Comment:
Don't do pseudo-one-liners like this. Use a proper `if` block. Also, you're introducing unnecessary subshell. Also, you're lacking error handling for all the lines below.